src/agents/regulator.jl

```diff
@@ -1515,12 +1598,13 @@ function solve_agent_problem!(
     end
     replace!(demand_cost_allocation_capacity_h, NaN => 0.0)
 
-    demand_cost_allocation_othercost_h = make_keyed_array(model_data.index_z, model_data.index_h)
+    demand_cost_allocation_othercost_h = make_keyed_array(model_data.index_z, model_data.index_h, regulator.index_cost_type)
     for (z,h) in model_data.index_z_h_map
-        demand_cost_allocation_othercost_h(z, h, :) .=
-            regulator.othercost(z, reg_year_index) * net_peak_load_wo_green_tech_h(z, h) / (
-                sum(net_peak_load_wo_green_tech_h(z, h) for h in z_to_h_dict[z])
-            )
+        demand_cost_allocation_othercost_h(z, h, :Distribution, :) .= calculate_demand_cost_allocation(regulator.othercost(z, reg_year_index, :Distribution), net_peak_load_wo_green_tech_h, z, h, z_to_h_dict)
+        demand_cost_allocation_othercost_h(z, h, :Transmission, :) .= calculate_demand_cost_allocation(regulator.othercost(z, reg_year_index, :Transmission), net_peak_load_wo_green_tech_h, z, h, z_to_h_dict)
+        demand_cost_allocation_othercost_h(z, h, :Interconnection, :) .= calculate_demand_cost_allocation(regulator.othercost(z, reg_year_index, :Interconnection), net_peak_load_wo_green_tech_h, z, h, z_to_h_dict)
+        demand_cost_allocation_othercost_h(z, h, :System, :) .= calculate_demand_cost_allocation(regulator.othercost(z, reg_year_index, :System), net_peak_load_wo_green_tech_h, z, h, z_to_h_dict)
+        demand_cost_allocation_othercost_h(z, h, :DERA, :) .= calculate_demand_cost_allocation(regulator.othercost(z, reg_year_index, :DERA), net_peak_load_wo_green_tech_h, z, h, z_to_h_dict)
```

Comment:
Instead of listing out the cost types, can you iterate over an index or plain vector?

Reply:
It seems like you just need an index that's a subset of `index_cost_type`?